### PR TITLE
rbd: fix spelling errors

### PR DESCRIPTION
**Pull request by jsoref**
This is based on https://github.com/ceph/ceph/pull/46824 and https://github.com/ceph/ceph/pull/46824

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

Note: I'm more than happy to squash, but I'd like to get buy-in for the changes first. Generally for changes w/ >10 things there's at least one that someone finds objectionable. And it's much easier for me to drop / fix them before I squash than after (and similarly, if I need to rebase, which I did for the parent PR, it's considerably more error prone if it's already squashed).

I'll annotate this PR shortly w/ the things that I think are most likely to be dropped/changed...

### CODE REVIEW

File: qa/tasks/rbd_fio.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def run_rbd_map(remote, image, iodepth):
 def run_fio(remote, config, rbd_test_dir):
     """
     create fio config file with options based on above config
-    get the fio from github, generate binary, and use it to run on
+    get the fio from GitHub, generate binary, and use it to run on
```
Comment (jsoref) [inline qa/tasks/rbd_fio.py:90]: brand

File: qa/workunits/windows/libvirt_vm/setup.sh [hunk 2]
```diff
@@ -77,7 +77,7 @@ sudo virt-install \
     --network network=default,model=virtio \
     --controller type=virtio-serial \
     --channel unix,target_type=virtio,name=org.qemu.guest_agent.0 \
-    --noautoconsol
+    --noautoconsole
```
Comment (jsoref) [inline qa/workunits/windows/libvirt_vm/setup.sh:80]: notable

File: src/librbd/cache/pwl/LogMap.cc [hunk 3]
```diff
@@ -250,7 +250,7 @@ LogMapEntries<T> LogMap<T>::find_map_entries_locked(const BlockExtent &block_ext
  * represent the entire block extent of the GenericWriteLogEntry, and the
  * WriteLogMapEntry is added to the set.
  *
- * The set must not contain overlapping WriteLogMapEntrys. WriteLogMapEntrys
+ * The set must not contain overlapping WriteLogMapEntries. WriteLogMapEntries
```
Comment (jsoref) [inline src/librbd/cache/pwl/LogMap.cc:253]: Probably wrong. Perhaps
```suggestion
 * The set must not contain overlapping `WriteLogMapEntry`s. `WriteLogMapEntry`s
```

File: src/tools/rbd_wnbd/rbd_wnbd.cc [hunk 4]
```diff
@@ -419,7 +419,7 @@ int map_device_using_suprocess(std::string arguments, int timeout_ms)
     case WAIT_OBJECT_0:
       if (!GetOverlappedResult(pipe_handle, &connect_o, &bytes_read, TRUE)) {
         err = GetLastError();
-        derr << "Couln't establish a connection with the child process. "
+        derr << "Couldn't establish a connection with the child process. "
```
Comment (jsoref) [inline src/tools/rbd_wnbd/rbd_wnbd.cc:422]: There was a request to expand the `mustn't` contraction. I didn't apply that throughout (I certainly can, but these PRs are already big enough, so I'd rather do it later...)

File: src/librbd/cache/pwl/LogMap.cc (hunk 3 above)
Comment (idryomov) [inline src/librbd/cache/pwl/LogMap.cc:253]: I think this spelling was intended.  I'd suggest the following:
```suggestion
 * The set must not contain overlapping write log entries. Entries
```

**Review by idryomov — COMMENTED** [review]
Please squash everything into a single commit -- 53 commits for 69 lines changed is really too much.

File: qa/workunits/windows/libvirt_vm/setup.sh (hunk 2 above)
Comment (idryomov) [reply qa/workunits/windows/libvirt_vm/setup.sh:80]: Yup -- libvirt must be accepting contracted options.

File: src/librbd/cache/pwl/LogMap.cc (hunk 3 above)
Comment (idryomov) [reply src/librbd/cache/pwl/LogMap.cc:253]: Looks like our comments crossed: I suggested skipping the class name entirely in another thread.

**Comment by idryomov** [comment]
jenkins test this please

**Review by idryomov — APPROVED** [review]

**Pull request merged**